services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/api_client/_public.py

```diff
@@ -1,27 +1,55 @@
 import logging
 from collections import deque
-from typing import Any, Optional
+from functools import cached_property
+from typing import Any, Final, Optional
 
 from fastapi import FastAPI, status
+from httpx import AsyncClient
 from models_library.projects import ProjectID
 from models_library.projects_networks import DockerNetworkAlias
-from pydantic import AnyHttpUrl
+from pydantic import AnyHttpUrl, PositiveFloat
+from servicelib.fastapi.long_running_tasks.client import (
+    Client,
+    ProgressCallback,
+    ProgressMessage,
+    ProgressPercent,
+    TaskId,
+    periodic_task_result,
+)
 from servicelib.utils import logged_gather
+from simcore_service_director_v2.core.settings import DynamicSidecarSettings
 
 from ....models.schemas.dynamic_services import SchedulerData
 from ....modules.dynamic_sidecar.docker_api import get_or_create_networks_ids
 from ....utils.logging_utils import log_decorator
-from ..errors import EntrypointContainerNotFoundError, NodeportsDidNotFindNodeError
+from ..errors import EntrypointContainerNotFoundError
 from ._errors import BaseClientHTTPError, UnexpectedStatusError
 from ._thin import ThinDynamicSidecarClient
 
+STATUS_POLL_INTERVAL: Final[PositiveFloat] = 1
+
 logger = logging.getLogger(__name__)
 
 
+async def _debug_progress_callback(
+    message: ProgressMessage, percent: ProgressPercent, task_id: TaskId
+) -> None:
+    logger.debug("%s: %.2f %s", task_id, percent, message)
+
+
 class DynamicSidecarClient:
     def __init__(self, app: FastAPI):
+        self._app = app
         self.thin_client: ThinDynamicSidecarClient = ThinDynamicSidecarClient(app)
 
+    @cached_property
+    def _async_client(self) -> AsyncClient:
+        return self.thin_client._client  # pylint: disable=protected-access
```

Comment:
why we still need to access the protected client of the thing_client? this does not feel correct

Reply:
Well I can make it public, since there no harm in that. We already wrap the thin_client. I will also make it private. Let's see if this feels better.

Reply:
no I still don't like it, but as discussed I guess I can live with it.